components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediatorFactory.java

```diff
@@ -0,0 +1,267 @@
+        }
+
+        CacheMediator cache = new CacheMediator();
+        OMAttribute idAttr = elem.getAttribute(ATT_ID);
+        String id;
+        //get cache store using the id of the mediator
+        if (idAttr != null && (id = idAttr.getAttributeValue()) != null) {
+            cacheStore = CacheStoreManager.get(id);
+            cache.setId(idAttr.getAttributeValue());
+        } else {
+            cacheStore = CacheStoreManager.get("");
+        }
+
+        cache.setCacheStore(cacheStore);
+
+        OMAttribute collectorAttr = elem.getAttribute(ATT_COLLECTOR);
+        if (collectorAttr != null && collectorAttr.getAttributeValue() != null &&
+                "true".equals(collectorAttr.getAttributeValue())) {
+            cache.setCollector(true);
+            cache.initPattern();
+        } else {
+            cache.setCollector(false);
+
+            OMAttribute timeoutAttr = elem.getAttribute(ATT_TIMEOUT);
+            if (timeoutAttr != null && timeoutAttr.getAttributeValue() != null) {
+                cache.setTimeout(Long.parseLong(timeoutAttr.getAttributeValue()));
+            } else {
+                cache.setTimeout(CachingConstants.DEFAULT_TIMEOUT);
+            }
+
+            OMAttribute maxMessageSizeAttr = elem.getAttribute(ATT_MAX_MSG_SIZE);
+            if (maxMessageSizeAttr != null && maxMessageSizeAttr.getAttributeValue() != null) {
+                cacheStore.setMaxMessageSize(Integer.parseInt(maxMessageSizeAttr.getAttributeValue()));
+            }
+
+            String className = null;
+            OMElement protocolElem = elem.getFirstChildWithName(PROTOCOL_Q);
+            Map<String, Object> props = new HashMap<>();
+            if (protocolElem != null) {
+                OMAttribute typeAttr = protocolElem.getAttribute(ATT_TYPE);
+
+                if (typeAttr != null &&
+                        typeAttr.getAttributeValue() != null) {
+                    OMElement hashGeneratorElem = protocolElem.getFirstChildWithName(HASH_GENERATOR_Q);
+                    if (hashGeneratorElem != null) {
+                        className = hashGeneratorElem.getText();
+                    }
+
+                    String protocolType = typeAttr.getAttributeValue().toUpperCase();
+                    cacheStore.setProtocolType(protocolType);
+                    if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(protocolType)) {
+                        OMElement methodElem = protocolElem.getFirstChildWithName(HTTP_METHODS_TO_CACHE_Q);
+                        if (methodElem != null) {
+                            String[] methods = methodElem.getText().split(",");
+                            if (!"".equals(methods[0])) {
+                                for (int i = 0; i < methods.length; i++) {
+                                    methods[i] = methods[i].trim();
+                                    if (!("POST".equals(methods[i]) || "GET".equals(methods[i]) || "HEAD".equals(
```

Comment:
I am checking these methods only in two places and even then I check all of them only in this place to validate. Do I need constants for all?